python/samples/getting_started_with_agents/multi_agent_orchestration/step5_magentic.py

```diff
@@ -0,0 +1,198 @@
+    StandardMagenticManager,
+)
+from semantic_kernel.agents.runtime import InProcessRuntime
+from semantic_kernel.connectors.ai.open_ai import OpenAIChatCompletion
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+    OpenAIChatPromptExecutionSettings,
+)
+from semantic_kernel.contents import ChatMessageContent
+
+"""
+The following sample demonstrates how to create a Magentic orchestration with two agents:
+- A Research agent that can perform web searches
+- A Coder agent that can run code using the code interpreter
+
+Read more about Magentic here:
+https://www.microsoft.com/en-us/research/articles/magentic-one-a-generalist-multi-agent-system-for-solving-complex-tasks/
+
+This sample demonstrates the basic steps of creating and starting a runtime, creating
+a Magentic orchestration with two agents and a Magentic manager, invoking the
+orchestration, and finally waiting for the results.
+
+The Magentic manager requires a chat completion model that supports structured output.
+"""
+
+
+async def agents() -> list[Agent]:
+    """Return a list of agents that will participate in the Magentic orchestration.
+
+    Feel free to add or remove agents.
+    """
+    research_agent = ChatCompletionAgent(
+        name="ResearchAgent",
+        description="A helpful assistant with access to web search. Ask it to perform web searches.",
+        instructions=(
+            "You are a Researcher. You find information without additional computation or quantitative analysis."
+        ),
+        # This agent requires the gpt-4o-search-preview model to perform web searches.
+        service=OpenAIChatCompletion(ai_model_id="gpt-4o-search-preview"),
+    )
+
+    # Create an OpenAI Assistant agent with code interpreter capability
+    client, model = OpenAIAssistantAgent.setup_resources()
+    code_interpreter_tool, code_interpreter_tool_resources = OpenAIAssistantAgent.configure_code_interpreter_tool()
+    definition = await client.beta.assistants.create(
+        model=model,
+        name="CoderAgent",
+        description="A helpful assistant with code interpreter capability.",
+        instructions="You solve questions using code. Please provide detailed analysis and computation process.",
+        tools=code_interpreter_tool,
+        tool_resources=code_interpreter_tool_resources,
+    )
+    coder_agent = OpenAIAssistantAgent(
+        client=client,
+        definition=definition,
+    )
+
+    return [research_agent, coder_agent]
+
+
+def agent_response_callback(message: ChatMessageContent) -> None:
+    """Observer function to print the messages from the agents."""
+    print(f"**{message.name}**\n{message.content}")
+
+
+async def main():
+    """Main function to run the agents."""
+    # 1. Create a Magentic orchestration with two agents and a Magentic manager
+    # Note, the Standard Magentic manager accepts custom prompts for advanced users and scenarios.
```

Comment:
those prompts are super specific, both in contents and in variables, so maybe good to call that out